sensor/common/compliance/nodescan_handler_test.go

```diff
@@ -131,14 +133,15 @@ func (s *NodeScanHandlerTestSuite) TestStopHandler() {
 				return
 			case nodeScans <- fakeNodeScanV2("Node"):
 				if i == 0 {
+					consumer.receivedC.Wait() // avoid race between consumer receiving the first message and hadler stopping
```

Comment:
You can achieve the same without introducing `receivedC` by just calling
```go
s.NoError(consumer.stoppedC.Wait())
```
instead of `consumer.receivedC.Wait()`.